src/app/modules/shared/components/breadcrumb/breadcrumb.component.ts

```diff
@@ -1,14 +1,22 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { BreadcrumbService } from '../../services/breadcrumb.service';
+import { ActivatedRouteSnapshot } from '@angular/router';
+import { switchMap } from 'rxjs/operators';
+import { EventsService } from 'src/app/modules/core/services/events.service';
 
 @Component({
   selector: 'app-breadcrumb',
   templateUrl: './breadcrumb.component.html',
 })
-export class BreadcrumbComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit(): void {
+export class BreadcrumbComponent {
+  constructor(
+    private breadcrumbService: BreadcrumbService,
+    private eventsService: EventsService,
+  ) {
   }
-
+  breadcrumbs$ = this.eventsService.routeChanged$.pipe(
```

Comment:
No need for subscriptions as we use the async pipe in the html file. This pipes the route changed events into the breadcrumb service create() function